src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java

```diff
@@ -206,4 +206,19 @@ private void sendAvailableNotice(CheckInProcessRecords records, Request firstReq
     }
     return succeeded(records);
   }
+
+  CheckInProcessRecords setInHouseUse(CheckInProcessRecords checkInProcessRecords) {
+    return checkInProcessRecords
+      .withInHouseUse(isInHouseUse(checkInProcessRecords));
+  }
+
+  private boolean isInHouseUse(CheckInProcessRecords records) {
+    if (records.getItem().getLocation() == null) {
+      return false;
+    }
```

Comment:
Under what circumstances would the location be null?

Should this decide that the check in is not in-house use, or instead, fail the check in because a location is needed for the logic to be appropriately applied?

Reply:
Well, the effective location can not be null, because holdings.permanentLocation is required, but it can be case when location object did not fetched and set to the item.

Reply:
> but it can be case when location object did not fetched and set to the item.

Why would the location not be fetched? Could it not exist or could it be a programming error?

What should the system do when that is the case? Should it assume that it is not in-house use, or fail?

Reply:
One of the possible cases, when someone decided to re-use the method but some dependencies won't be fetched (since we're allowing such behavior). 

I think that we should not fail check-in, because in-house info is not critical part of the transaction.

Reply:
> One of the possible cases, when someone decided to re-use the method but some dependencies won't be fetched (since we're allowing such behavior).

And I think at that point, it is better to fail, so the developer can correct that situation early.

> I think that we should not fail check-in, because in-house info is not critical part of the transaction.

I think that is a decision for Emma. Personally, I think it is safer to fail the check in.

Reply:
Will it work for you if I add logging for this case instead of failing?

Reply:
> Will it work for you if I add logging for this case instead of failing?

By all means add logging, personally I don't think it adds much because it doesn't raise provoke investigation like failing does.

Reply:
Ok, I'll remove this check then, because I don't know how to recreate this case using integration tests (it is not enough to just unit test, because we have to make sure that the failure is correctly propagated through app and returned in the response).